qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -74,6 +108,11 @@ def __init__(
         self._points_optparams: Dict[float, List[float]] = None
         self._num_bootstrap = num_bootstrap
         self._extrapolator = extrapolator
+        self._solver = None
+        if isinstance(self._gss, GroundStateSolver):
+            self._solver = self._gss.solver
```

Comment:
Saved solver in a member in order to have the same code for both implementations (GroundStateSolver and MinimalEigensolverFactory). 
GroundStateSolver has the solver in the instance, MinimalEigensolverFactory produces a solver.